### PR TITLE
feat(settings): use JSON editor for MCP servers + compact one-line list

### PR DESCRIPTION
**Pull request by gnoviawan**
## Summary

Application Preferences → MCP Servers still used bulky form input fields for adding/editing a server — the raw-JSON editing requested previously only shipped as an add-only "Import JSON" side tab — and each server rendered as a multi-line card. This PR makes the JSON editor THE Add/Edit interface and compacts the list to one line per server, so users can paste/edit Claude-Desktop-style configs directly while entries still persist through the existing registry list (same storage format, desktop + web).

## Related Issue

Refs #362 (follow-up UX improvement to the MCP Servers settings surface introduced there)

## Type of Change

- [x] feat: new feature
- [ ] fix: bug fix
- [ ] docs: documentation only
- [ ] style: formatting or non-functional styling changes
- [ ] refactor: internal cleanup with no behavior change
- [ ] perf: performance improvement
- [ ] test: adds or updates tests
- [ ] build: build or dependency changes
- [ ] ci: CI/CD workflow changes
- [ ] chore: maintenance or tooling
- [ ] revert: reverts a previous change

## What Changed

- `src/renderer/components/settings/McpServersSettings.tsx`
  - Add/Edit dialog body is now a raw JSON editor (monospace textarea). Add mode accepts a Claude Desktop `{"mcpServers": {...}}` wrapper or a single server object and saves each parsed entry with a fresh id; Edit mode pre-fills that server's JSON (`{type, name, command|url, args, env-map|headers-pairs, enabled}`), rejects the wrapper, honors an explicit `enabled` override, and updates the same registry id.
  - Server list compacted from multi-line cards to one line per server: status dot, name, transport badge, command/URL, tools trigger, Test, enable switch, edit, delete — with tool/probe details behind the existing collapsible disclosure.
  - Removed the old form fields, the Form/Import tab bar, and the now-unused draft helpers.
- `src/renderer/components/settings/McpServersSettings.test.tsx` — rewritten suite (30 tests) covering every edge case: add bare object / wrapper / invalid JSON / mixed validity / empty wrapper; edit round-trip with unknown-field drop, HTTP headers pre-fill, `enabled` override + omission, wrapper rejection, validation errors; save-failure rollback for both add and edit; plus preserved behaviors (toggle/delete, probe dots, probe-error disclosure, tool list, Test button, compact-row assertions).
- `src/renderer/assets/agent-icons/acp/agents.json` — re-synced bundled ACP registry snapshot with the CDN (required by the ACP Registry Bundle Freshness gate; version bumps only).

Storage schema, store actions, probes, chatbox badge, facades, and desktop↔server parity are untouched (UI-only change; both desktop and web entries render this same component). Reuses the existing `parseMcpJsonImport` unchanged.

## How It Was Tested

- [x] `bun run ci` (Biome lint + format + imports, strict mode)
- [x] `bun run typecheck`
- [x] `bun run test`
- [ ] `cargo clippy --all-targets -- -D warnings` (in src-tauri)
- [ ] `cargo test` (in src-tauri)
- [ ] Manual verification completed
- [ ] Not applicable

(`bun run ci`, `bun run typecheck`, and `bun run test` all pass locally — 251 files / 3021 tests passed, 43 skipped. No `src-tauri` changes, so Rust gates are re-validated by CI.)

## Screenshots or Recordings

Row layout: `[dot] Name STDIO command… [n tools ▾] [test] [switch] [edit] [delete]`, all on one line. Add/Edit opens a JSON-only dialog with inline validation errors. (Screenshot on request.)

## CI & Review Gate

- [ ] All CI checks pass (PR Validation, Rust Checks, Build Verification, Security Scans)
- [ ] Code review comments from CodeRabbit / Claude have been addressed or resolved
- [ ] No unresolved review findings remain

## Checklist

- [x] My PR title follows the conventional commit format used by this repo
- [x] I linked the related issue or explained why none exists
- [x] I updated docs when needed
- [x] I added or updated tests when needed
- [x] I verified the change does not introduce unrelated modifications
- [x] I read AGENTS.md and followed the contributor guidelines
- [x] A human reviewed the complete diff before submission

## Summary by CodeRabbit

* **New Features**
  * Redesigned MCP server setup and editing with a JSON-based workflow.
  * Added inline validation, clearer error feedback, and Claude Desktop wrapper imports.
  * Added compact, expandable server entries with transport details, probe status, available tools, and actions.
  * Added atomic batch imports with rollback if saving fails.

* **Bug Fixes**
  * Prevented overlapping MCP server changes from overwriting one another.
  * Preserved server enabled states during edits and saves.

* **Updates**
  * Updated the catalog of available ACP agents and distributions, including Harn and Kilo.

### CODE REVIEW

**Comment by gnoviawan** [comment]
@coderabbitai review

File: src/renderer/components/settings/McpServersSettings.tsx [hunk 1]
```diff
@@ -155,69 +98,112 @@ export function McpServersSettings(): React.JSX.Element {
...
-      // Tell the user how many landed so they do not blindly re-paste and
+      // Tell the user how many landed so they do not blindly re-save and
       // create duplicates.
       toast.error(
         savedCount > 0
-          ? `Imported ${savedCount} server${savedCount === 1 ? '' : 's'} before a save failed. Review the remaining entries before re-importing.`
+          ? `Imported ${savedCount} server${savedCount === 1 ? '' : 's'} before a save failed. Review the remaining entries before re-saving.`
           : 'Could not save the MCP server. Your previous settings were restored.'
       )
+    }
```
[the remark this answers is not in the document]
Comment (gnoviawan) [reply src/renderer/components/settings/McpServersSettings.tsx]: Fixed in b00a0da8. The add flow is now all-or-nothing: the whole input is validated via parseMcpJsonImport before anything persists, and the accepted batch commits through a single atomic importMcpServers store write. A corrected re-save therefore starts from the same registry state and cannot duplicate earlier entries. Added a retry-no-duplicate test (McpServersSettings.test.tsx) and atomic-persist/rollback tests (acp-store.test.ts).

File: src/renderer/components/settings/McpServersSettings.tsx [hunk 2]
```diff
@@ -155,69 +98,112 @@ export function McpServersSettings(): React.JSX.Element {
...
-      // Valid servers still import when some entries are rejected (per-server
-      // errors stay visible inline so the user can fix and re-paste).
-      for (const server of parsedServers) {
-        await saveMcpServer({ ...server, id: crypto.randomUUID(), enabled: true })
+      const { servers: parsedServers, errors } = parseMcpJsonImport(jsonText)
+      // Valid servers still save when some entries are rejected (per-server
+      // errors stay visible inline so the user can fix and re-save).
+      for (const parsed of parsedServers) {
+        await saveMcpServer({ ...parsed, id: crypto.randomUUID(), enabled: true })
         savedCount += 1
```
[the remark this answers is not in the document]
Comment (gnoviawan) [reply src/renderer/components/settings/McpServersSettings.tsx]: Fixed in b00a0da8. Added importMcpServers(servers[]) to acp-store.ts — one optimistic update + one disk write + rollback. saveAdd now accumulates the full batch (fresh id + enabled per entry) and calls it once, so a wrapper import triggers a single registry write and one probe-shape change instead of one per server.

File: src/renderer/components/settings/McpServersSettings.tsx [hunk 3]
```diff
@@ -426,188 +401,55 @@ export function McpServersSettings(): React.JSX.Element {
       </div>
 
       <Dialog
-        open={draft !== null || importMode}
+        open={dialog !== null}
         onOpenChange={(open) => {
           if (!open) closeDialog()
         }}
```
[the remark this answers is not in the document]
Comment (gnoviawan) [reply src/renderer/components/settings/McpServersSettings.tsx:408]: Fixed in b00a0da8. Added a !saving guard to the Dialog onOpenChange and disabled Cancel while saving, so Escape/overlay/Cancel cannot dismiss mid-save; the in-flight completion handler then settles normally. Added a dismissal-guard regression test using a deferred save promise.

**Comment by gnoviawan** [comment]
@coderabbitai review

File: src/renderer/stores/acp-store.ts [hunk 4]
```diff
@@ -4110,6 +4117,23 @@ export const useAcpStore = create<AcpState>((set, get) => ({
...
+      await saveMcpServersToDisk(next)
+    } catch (err) {
+      set({ mcpServers: list })
+      void logFrontendError({
+        source: 'acp-store.importMcpServers',
+        message: `Failed to persist MCP registry import (${String(err)})`
+      })
+      throw err
+    }
+  },
```
[the remark this answers is not in the document]
Comment (gnoviawan) [reply src/renderer/stores/acp-store.ts:4154]: Fixed in 87016cc4. Added a module-level promise queue (
unSerializedMcpRegistryMutation) that all four MCP registry mutations (saveMcpServer, importMcpServers, setMcpServerEnabled, deleteMcpServer) now run through. Each mutation computes the next list, persists, and — on failure — rolls back within its serialized turn, so an overlapping mutation can neither persist a stale snapshot after a newer one nor have its rollback drop an intervening change. Added a regression test that holds an import write in flight while a toggle is issued and asserts both writes land in mutation order with the toggle's snapshot including the imported entry (acp-store.test.ts).

**Comment by gnoviawan** [comment]
@coderabbitai review

**Comment by gnoviawan** [comment]
@coderabbitai review
